rfcs/rfc-006-budget-contract-language.md

```diff
@@ -0,0 +1,117 @@
+
+### Transfer (aka, Budget v2) Transaction Format
+```rust
+pub enum TransferCondition {
```

Comment:
This was called Condition in budget.rs, which seems like a better name, because it doesn't assume the consequence of satisfying that condition would be to make a payment. Note that `Budget` is an enum with `After(Condition, Payment)`. The only reason that's not `After(Condition, Box<Budget>)` is because we didn't write the static analysis code to ensure that's not a really big, expensive expression.